nodebuilder/tests/nd_test.go

```diff
@@ -48,8 +49,12 @@ func TestShrexNDFromLights(t *testing.T) {
 	// wait for chain to be filled
 	require.NoError(t, <-fillDn)
 
-	// first 2 blocks are not filled with data
-	for i := 3; i < blocks; i++ {
+	// first 15 blocks are not filled with data
+	//
+	// TODO: we need to stop guessing
+	// the block that actually has transactions. We can get this data from the
+	// response returned by FillBlock.
+	for i := 16; i < blocks; i++ {
```

Comment:
I think we should change the channel returned by `FillBlocks` to return a channel of `sdk.TxResult` instead of error. This way, we can actually see which blocks the txs get included in and are not making crude trial and error estimates on which blocks have data in them.

If needed, we can make it return some summary of what was submitted as well if that would be useful.

Reply:
This change may collide with some changes @walldiss is working on to make tests with filled blocks more deterministic